backends/s3/s3.go

```diff
@@ -93,10 +96,11 @@ type Options struct {
 }
 
 func (o Options) Check() error {
-	if o.AccessKey == "" {
+	hasProvider := o.Provider != K8sSecretProvider{}
```

Comment:
As a result of directly embedding the Provider in the config struct, you cannot do additional initialization and safely add private fields without breaking these config checks. Best to separate the configuration from the implementation here.